### ISSUE TITLE
Redefinition of signals

### ISSUE THREAD

**Issue by danielwilms** [COVESA/vehicle_service_catalog#2]
> - name: position_t
          type: struct
          description: | 
            The position of the entire seat
          members:
            - name: base
              type: movement.movement_t
              description: |
                The position of the base 0 front, 1000 back
            - name: cushion
              type: movement.movement_t
              description: |
                The position of the cushion 0 short, 1000 long
            - name: lumbar
              type: movement.movement_t
              description: |
                The position of the lumbar support
            - name: side_bolster
              type: movement.movement_t
              description: |
                The position of the side bolster
            - name: head_restraint
              type: movement.movement_t
              description: |
                The position of the head restraint 0 down, 1000 up

@gunnarx, @magnusfeuer I'm really concerned about yet another definition of vehicle concepts in the interface description throughout the seat example. Why aren't you using VSS, where possible?

**Comment by UlfBj** [comment]
>> Why aren't you using VSS, where possible?

I second @danielwilms view on this. 
Duplication is not a good design choice.

**Comment by adobekan** [comment]
Would be interesting to understand main differences, and why not aligning to VSS?

**Comment by gunnarx** [comment]
I was considering closing this old ticket but we should document where we are and how we see things.
Some of the underlying concerns might still be valid so the ticket might also be kept open unless we cover those aspects somewhere else.

A few things to consider:

> @gunnarx, @magnusfeuer I'm really concerned about yet another definition of vehicle concepts in the interface description throughout the seat example

- Most effort is currently on finalizing the features of the VSC interface definition language, define its relationship to existing IDLs (Franca, OpenAPI, Protobuf/gRPC and others), and get input from the industry on what they prefer and thus how to build a shared strategy around interface definition.  
- Therefore the seats service / comfort service can be seen as only an example at this point**, It is used to drive the definition of the language and there will be time in the future to align the signal (VSS) and services  (VSC) catalogs to avoid overlap.
- **More / other interface examples are welcome!  If you provide them, the problem might go away for now.
- We still have some remaining discussion on the best ways to integrate VSS signals and VSC interfaces.  We definitely want to get a very good integration.

> Would be interesting to understand main differences, and why not aligning to VSS?

"main differences" I think is related to the purpose/usage of VSC and VSS.  Here are a few aspects that have come up:  
- Some propose that actuation should _primarily_ happen through a method call instead of writing a VSS signal.  The motivation and assumption is that an RPC protocol is likely to have a richer way to report progress, errors, and completion. 
- Others say that writable (actuator) VSS signals and the typical protocols we now have (VISS, and others with similar capability) are acceptable for triggering functions to happen. 
- The example brought up here  (controlling seat position) might be a perfect example to discuss this difference in opinion.
- To my understanding, we have not reached consensus on this point, so I conclude that the _capability_ of VSS and VSC should at least prepare for doing it in either one way or the other.
- In the meantime, in a VSS meeting we agreed that setting an actuator value should generally be considered as setting the "target" value (but I didn't find it in the documentation??)
  - In other words, it is assumed that if this were to be used for something like "set window roll-down position to 50%" then typical protocols (e.g. VISS) are expected to return a success for writing the value and end the write operation, even while it might take additional time for the window to reach that position.  (Checking the status/success of the operation would require monitoring signals such as the current window position).  This clarification might not solve the entire challenge of the actuation interface, but it makes the expectation of behavior less ambiguous.
